File: DSA/Cpp/src/dsa/lib/lists/arraylist.cpp
#include "arraylist.h"

namespace dsa
{
    namespace lib
    {
        ArrayList::ArrayList()
        {
            int array[DEFAULT_LENGTH];
            this->lst = array;
        };
        ArrayList::ArrayList(int ary[]) { this->lst = ary; };
        ArrayList::~ArrayList() { delete[] lst; }

    } // namespace lib
} // namespace dsa

File: DSA/Java/src/dsa/lib/lists/ArrayList.java
package dsa.lib.lists;

public class ArrayList implements List {
    private static final int DEFAULT_LENGTH = 10;
    private final int[] lst;

    public ArrayList() {
        this.lst = new int[DEFAULT_LENGTH];
    }

    @Override
    public int size() {
        return 0;
    }
}
